packages/filebrowser-extension/src/index.ts

```diff
@@ -254,7 +255,15 @@ const browser: JupyterFrontEndPlugin<void> = {
             showHiddenFiles = settings.get('showHiddenFiles')
               .composite as boolean;
 
-            browser.showHiddenFiles = showHiddenFiles;
+            settings.changed.connect(settings => {
+              filterDirectories = settings.get('filterDirectories')
+                .composite as boolean;
+              browser.model.filterDirectories = filterDirectories;
+            });
+            filterDirectories = settings.get('filterDirectories')
+              .composite as boolean;
+
+            browser.model.filterDirectories = filterDirectories;
```

Comment:
Sounds good. Either we can do it in this PR. Otherwise happy to track it separately as a 4.0 item but should be done before release, since it is likely to be breaking change.